src/custom/sdk/custom_instance.h

```diff
@@ -73,7 +74,20 @@ class CustomInstance {
   /// output (see CustomGetOutputFn_t).
   virtual int Execute(
       const uint32_t payload_cnt, CustomPayload* payloads,
-      CustomGetNextInputFn_t input_fn, CustomGetOutputFn_t output_fn) = 0;
+      CustomGetNextInputFn_t input_fn, CustomGetOutputFn_t output_fn) { return -1;}
+
+  /// Execute the custom instance. User should override this function
+  /// if version 2 of the custom interface is used.
+  ///
+  /// \param payload_cnt The number of payloads to execute.
+  /// \param payloads The payloads to execute.
+  /// \param input_fn The callback function to get tensor input (see
+  /// CustomGetNextInputVer2Fn_t).
+  /// \param output_fn The callback function to get buffer for tensor
+  /// output (see CustomGetOutputVer2Fn_t).
+  virtual int Execute(
```

Comment:
Add
"\return Error code indicating success or the type of failure"
Instead of returning -1 (which is an non-defined error code) you should add a predefined error code to error_codes.h and return that . The description for that error code should be something about how a V2 version of a function was called when custom backend is not V2.